lib/modules/manager/haskell-cabal/extract.spec.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  countPackageNameLength,
+  countPrecedingIndentation,
+  extractNamesAndRanges,
+  findExtents,
+  splitSingleDependency,
+} from './extract';
+
+describe('modules/manager/haskell-cabal/extract', () => {
+  describe('countPackageNameLength', () => {
+    it.each`
+      input    | expected
+      ${'-'}   | ${null}
+      ${'-j'}  | ${null}
+      ${'-H'}  | ${null}
+      ${'j-'}  | ${null}
+      ${'3-'}  | ${null}
+      ${'-3'}  | ${null}
+      ${'3'}   | ${null}
+      ${'æ'}   | ${null}
+      ${'æe'}  | ${null}
+      ${'j'}   | ${1}
+      ${'H'}   | ${1}
+      ${'0ad'} | ${3}
+      ${'3d'}  | ${2}
+    `('matches $input', ({ input, expected }) => {
+      const maybeIndex = countPackageNameLength(input);
+      expect(maybeIndex).toStrictEqual(expected);
+    });
+  });
+
+  describe('countPrecedingIndentation()', () => {
+    it.each`
+      content                                       | index | expected
+      ${'\tbuild-depends: base\n\tother-field: hi'} | ${1}  | ${1}
+      ${' build-depends: base'}                     | ${1}  | ${1}
+      ${'a\tb'}                                     | ${0}  | ${0}
+      ${'a\tb'}                                     | ${2}  | ${1}
+      ${'a b'}                                      | ${2}  | ${1}
+      ${'  b'}                                      | ${2}  | ${2}
+    `(
+      'countPrecedingIndentation($content, $index)',
+      ({ content, index, expected }) => {
+        expect(countPrecedingIndentation(content, index)).toBe(expected);
+      },
+    );
+  });
+
+  describe('findExtents()', () => {
+    it.each`
+      content                | indent | expected
+      ${'a: b\n\tc: d'}      | ${1}   | ${10}
+      ${'a: b'}              | ${2}   | ${4}
+      ${'a: b\n\tc: d'}      | ${2}   | ${4}
+      ${'a: b\n '}           | ${2}   | ${6}
+      ${'a: b\n c: d\ne: f'} | ${1}   | ${10}
+    `('findExtents($indent, $content)', ({ indent, content, expected }) => {
+      expect(findExtents(indent, content)).toBe(expected);
+    });
+  });
+
+  describe('splitSingleDependency()', () => {
+    it.each`
+      depLine              | expectedName | expectedRange
+      ${'base >=2 && <3'}  | ${'base'}    | ${'>=2 && <3'}
+      ${'base >=2 && <3 '} | ${'base'}    | ${'>=2 && <3'}
+      ${'base>=2&&<3'}     | ${'base'}    | ${'>=2&&<3'}
+      ${'base'}            | ${'base'}    | ${''}
```

Comment:
Add a test which returns null

Reply:
```suggestion
      ${'base'}            | ${'base'}    | ${''}
      ${'  test'}          | ${null}      | ${null}
```

Reply:
There is already a test for the case where `splitSingleDependency` returns null. It is below, on line 83, search for `-invalid-package-name`.  The issue with the proposed line `${'  test'}          | ${null}      | ${null}` is that it doesn't pass, since `toEqual` is used, and the `res?.name` lookup makes it return `undefined`. So I thought it would be better to test this separately.